test/basic.jl

```diff
@@ -396,6 +396,43 @@ end
     # get_view_compiled = @compile get_view(x_concrete)
 end
 
+function masking(x)
+    y = similar(x)
+    y[1:2, :] .= 0
+    y[3:4, :] .= 1
+    return y
+end
+
+function masking!(x)
+    x[1:2, :] .= 0
+    x[3:4, :] .= 1
+    return x
+end
+
+@testset "setindex with views" begin
```

Comment:
sorry for being annoying, but apparently there is another function called `Base.setindex` used in `Tuple`s

```suggestion
@testset "setindex! with views" begin
```